src/core/router/info.h

```diff
@@ -492,14 +480,17 @@ class RouterInfo : public RouterInfoTraits, public RoutingDestination
     }
   };
 
+  /// @brief Check if the passed address already exist.
+  ///        This function will check if it has a compatible host and transport.
+  /// @params Address to look up for.
+  /// @params return true if the address is present.
+  bool ExistAddress(const Address& addr) const;
+
   /// @brief Adds/saves address + sets appropriate RI members
-  /// @param point Supported transport / Host string / Port integral
-  /// @param key Our intoducer key
-  /// @param mtu Address MTU
-  void AddAddress(
-      const std::tuple<Transport, std::string, std::uint16_t>& point,
-      const std::uint8_t* key = nullptr,
-      const std::uint16_t mtu = 0);
+  /// @params addr rvalue (will be moved into our container).
+  /// @params add_key If true we will insert the key using the IdentityEx
+  ///         which was passed during construction.
+  void AddAddress(Address&& addr, const bool add_key = false);
```

Comment:
This won't scale, I just noticed this.  If keep the interface, it should be a template function and take an universal reference to guarantee it will work either with rvalues or lvalues(and all the flavors)